utilities/test_suite/rpp_test_suite_common.h

```diff
@@ -0,0 +1,414 @@
+
+inline void remove_substring(string &str, string &pattern)
+{
+    std::string::size_type i = str.find(pattern);
+    while (i != std::string::npos)
+    {
+        str.erase(i, pattern.length());
+        i = str.find(pattern, i);
+   }
+}
+
+template <typename T>
+inline void compare_output(T* output, string funcName, RpptDescPtr srcDescPtr, RpptDescPtr dstDescPtr, RpptROI *roiPtr, int noOfImages)
+{
+    bool isEqual = true;
+    string func = funcName;
+    string refPath = get_current_dir_name();
+    string pattern = "/build";
+    remove_substring(refPath, pattern);
+    string dataType[4] = {"_u8_", "_f16_", "_f32_", "_i8_"};
+
+    if(srcDescPtr->dataType == dstDescPtr->dataType)
+        func += dataType[srcDescPtr->dataType];
+    else
+    {
+        func = func + dataType[srcDescPtr->dataType];
+        func.resize(func.size() - 1);
+        func += dataType[dstDescPtr->dataType];
+    }
+
+    if(dstDescPtr->layout == RpptLayout::NHWC)
+        func += "Tensor_PKD3";
+    else
+    {
+        if (dstDescPtr->c == 3)
+            func += "Tensor_PLN3";
+        else
+            func += "Tensor_PLN1";
+    }
+
+    string refFile = refPath + "/../REFERENCE_OUTPUT/" + funcName + "/"+ func + ".csv";
+    ifstream file(refFile);
+    Rpp8u *refOutput;
+    refOutput = (Rpp8u *)malloc(noOfImages * dstDescPtr->strides.nStride * sizeof(Rpp8u));
+    string line,word;
+    int index = 0;
+
+    // Load the refennce output values from files and store in vector
+    if(file.is_open())
+    {
+        while(getline(file, line))
+        {
+            stringstream str(line);
+            while(getline(str, word, ','))
+            {
+                refOutput[index] = stoi(word);
+                index++;
+            }
+        }
+    }
+    else
+    {
+        cout<<"Could not open the reference output. Please check the path specified\n";
+        return;
+    }
+
+    for(int c = 0; c < noOfImages; c++)
+    {
+        Rpp8u *outputTemp = output + c * dstDescPtr->strides.nStride;
+        Rpp8u *outputTempRef = refOutput + c * dstDescPtr->strides.nStride;
+        for(int i = 0; i < roiPtr->xywhROI.roiHeight; i++)
+        {
+            Rpp8u *rowTemp = outputTemp + i * dstDescPtr->strides.hStride;
+            Rpp8u *rowTempRef = outputTempRef + i * dstDescPtr->strides.hStride;
+            for(int j = 0; j < roiPtr->xywhROI.roiWidth; j++)
+            {
+                Rpp8u *outVal = rowTemp + j;
+                Rpp8u *outRefVal = rowTempRef + j;
+                int diff = abs(*outVal - *outRefVal);
+                if(diff > CUTOFF)
+                {
+                    isEqual = false;
+                    break;
+                }
+            }
+        }
+    }
+    if(isEqual == true)
+        cout<< func << ": " << "PASSED \n";
+    else
+        cout<< func << ": " << "FAILED \n";
+}
+
+inline void write_image(string outputFolder, Rpp8u *output, RpptDescPtr dstDescPtr, string imageNames[], RpptImagePatch *dstImgSizes)
```

Comment:
Lets call this function write_image_batch_opencv(), and similarly modularize for the read too with a funciton called read_image_batch_opencv().
In the future, for the next set of changees, the rpp test suite should take an enum for decoder type - opencv/libjpeeg/turbojpeg or other. Pls design accordingly
```
if (decoderType = rppTestSuiteDecoder::opencv)
    read_image_batch_opencv();
else if (decoderType = rppTestSuiteDecoder::turbojpeg)
    read_image_batch_turbojpeg();
...
```

Reply:
Done